toml/src/main/kotlin/org/rust/toml/completion/CargoTomlDependencyFeaturesCompletionProvider.kt

```diff
@@ -35,9 +37,12 @@ class CargoTomlDependencyFeaturesCompletionProvider : CompletionProvider<Complet
         val pkgName = containingArray.containingDependencyKey?.text ?: return
 
         val depToml = findDependencyTomlFile(containingArray, pkgName) ?: return
+        val originalArray = CompletionUtil.getOriginalElement(containingArray) ?: return
         // TODO avoid AST loading?
         for (feature in depToml.allFeatures()) {
-            result.addElement(lookupElementForFeature(feature))
+            if (originalArray.elements.none { it.stringValue == feature.name }) {
+                result.addElement(lookupElementForFeature(feature))
```

Comment:
nit: there is `com.intellij.codeInsight.completion.CompletionResultSet#addAllElements`